go-controller/pkg/ovn/obj_retry.go

```diff
@@ -742,6 +800,13 @@ func (oc *Controller) iterateRetryResources(r *retryObjs, updateAll bool) {
 		if entry.ignore {
 			continue
 		}
+		// storing original obj for metrics
+		var initObj interface{}
+		if entry.newObj != nil {
+			initObj = entry.newObj
+		} else if entry.oldObj != nil {
+			initObj = entry.oldObj
+		}
```

Comment:
OK, I understand now why you recently asked me if both newObj and oldObj can be `nil`. If you get an update on an object and you already have a retry entry for this object marked for deletion (i.e. its oldObj != nil), then deletion might fail, in which case you will end up with a retry entry with oldObj!= nil (it's the old instance you were trying to delete) and `newObj!=nil` (it's the new instance from the update event). The same goes for an add event on an object with an existing retry entry marked for deletion, if deletion fails.

Anyway, here I see that the whole purpose of `initObj` is to pass it to `recordSuccessEvent`, from which you extract namespace and object name, which are not allowed by the API server to vary across an update, that is, in retry terms,  between `oldObj` and `newObj`. So in the end this if/else block is OK as it is! :)